proxy-injector/inject/inject.go

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package inject
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-mesh/openlogging"
+
+	"github.com/apache/servicecomb-mesher/proxy-injector/templates"
+	"k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (wh *Webhook) inject(review *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	reqBytes := review.Request.Object.Raw
+	pod := &corev1.Pod{}
+	if err := json.Unmarshal(reqBytes, pod); err != nil {
+		openlogging.Error("json marshal pod: %s", openlogging.WithTags(openlogging.Tags{
+			"err": err.Error(),
+		}), openlogging.WithTags(openlogging.Tags{
+			"bytes": string(reqBytes),
+		}))
+		return responseFromErrorMessage(err.Error())
+	}
+
+	tmpl, err := templates.NewTemplater(wh.config.templateName, wh.config.sidecarConfig, wh.config.sidecarTemplate)
+	if err != nil {
+		openlogging.Error(fmt.Sprintf("new templater failed: error = %v, template = %s", err.Error(), wh.config.templateName))
+		return responseFromErrorMessage(err.Error())
+	}
+
+	tmplSpec := tmpl.PodSpecFromTemplate(pod)
+	if tmplSpec == nil {
+		openlogging.Error("get pod spec from template failed")
+		return responseFromErrorMessage("get pod spec from template failed")
+	}
+
+	tmplSpec.Containers = addK8sServiceAccount(pod.Spec.Containers, tmplSpec.Containers)
+
+	pathBytes, err := createJsonPatch(&pod.Spec, tmplSpec)
+	if err != nil {
+		openlogging.Error("create patch failed: " + err.Error())
+		return responseFromErrorMessage(err.Error())
+	}
+
+	return &v1beta1.AdmissionResponse{
+		Allowed: true,
+		Patch:   pathBytes,
+		PatchType: func() *v1beta1.PatchType {
+			pt := v1beta1.PatchTypeJSONPatch
+			return &pt
+		}(),
+	}
+}
+
+func addK8sServiceAccount(dst, src []corev1.Container) []corev1.Container {
+	var serviceAccount *corev1.VolumeMount
+found:
```

Comment:
Why use found: and next: ?

Reply:
Never use goto block

Reply:
> Why use found: and next: ?

In the block,
used "break found" is to terminate multiple level loops,
used "continue next" is to skip the first level loop.

Reply:
do not use goto statement